### PR TITLE
fix agg bug in some corner cases

### LINKED ISSUE #8799: type: std::out_of_range, e.what() = vector
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

[reproduce.txt](https://github.com/pingcap/tiflash/files/14428039/reproduce.txt)
```sql
select count( distinct "20") from table_1_utf8mb4collate_utf8mb4_general_ci_undef as t1 where t1. `col_char(20)_key_signed` = all (select `col_char(20)_key_signed` from table_20_utf8mb4collate_utf8mb4_bin_undef as t2 where "YmkS" !=  `col_char(20)_key_signed`);
-- ERROR 1105 (HY000): other error for mpp stream: std::exception. Code: 1001, type: std::out_of_range, e.what() = vector
```
### 4. What is your TiFlash version? (Required)
master

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: close #8799

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Interpreters/Aggregator.cpp [hunk 1]
```diff
@@ -226,6 +226,11 @@ Block Aggregator::getHeader(bool final) const
     return params.getHeader(final);
 }
 
+Block Aggregator::getSourceHeader() const
+{
+    return params.src_header;
+}
```
Comment (SeaRise) [inline dbms/src/Interpreters/Aggregator.cpp:232]: why need this？
how about adding some comments here?

Comment (windtalker) [reply dbms/src/Interpreters/Aggregator.cpp:235]: Done

**Review by SeaRise — APPROVED** [review]
lgtm

**Review by yibin87 — APPROVED** [review]
LGTM

**Pull request merged**